Update libtock-rs to the latest master as of PR 112.

jrvanwhy opened the pull request:
Due to the newly-added futures support in libtock-rs, this required me to vendor several large crates. I am working on a solution to that issue, but this will have to do for now. Additionally, our build system now builds libtock-rs' examples (which brings in a few more dependencies).

Crates added:

1. **byteorder-1.2.7:** Dependency of corepack. Byteorder 0.* is already a dependency of elf2tab.
1. **corepack:** Required to build libtock-rs' examples (listed in `[dev-dependencies]`). Needed because I added libtock-rs' examples to our build system.
1. **futures-channel:** Depedency of futures.
1. **futures-core:** Dependency of futures.
1. **futures-io:** Dependency of futures.
1. **futures-sink:** Dependency of futures.
1. **futures-task:** Dependency of futures.
1. **futures-util:** Dependency of futures.
1. **futures:** New dependency of libtock-rs.
1. **pin-utils:** Dependency of futures and futures-util.
1. **proc-macro2-1.0.6:** Dependency of libtock-rs' codegen crate. elf2tab already pulls in 0.4.6 so we need both.
1. **quote-1.0.2:** Dependency of libtock-rs' codegen crate.
1. **serde:** Dependency of libtock-rs' codegen crate.
1. **serde_derive:** Dependency of corepack and serde.
1. **syn-1.0.11:** Dependency of libtock-rs' codegen crate.
1. **unicode-xid-0.2.0:** Dependency of proc-macro2-1.0.6 and syn-1.0.11.

```
----------------------
`make prtest` summary:
----------------------
git rev-parse HEAD
abcb200cdcbe6c0fcfda3d69efd60ed90ac522ae
git status
On branch libtock-rs-update
Your branch is up to date with 'origin/libtock-rs-update'.

nothing to commit, working tree clean
```

Commit:
Update libtock-rs to ab2c945, which includes PR 95. This brings our libtock-rs commit back to a previous mainline commit.

Commit:
Update to libtock-rs 8e96724.

Commit:
Add tweaks to support libtock-rs's async timer driver.

Commit:
Update libtock-rs to the current revision of libtock-rs PR 114.

I also enabled the building of libtock-rs' examples in tock-on-titan, to allow me to test PRs without relying on Travis. Unfortunately this update involved a lot of vendoring, both for the examples and for futures-rs.

Commit:
Update to libtock-rs as of PR 112.

phil-levis-google commented:
I went through the crates and simultaneously think it makes sense they're needed and think it's a lot of code!

Just for recording purposes, could you list the crates and a brief summary of their purpose/role in your PR?

jrvanwhy commented:
> I went through the crates and simultaneously think it makes sense they're needed and think it's a lot of code!
> 
> Just for recording purposes, could you list the crates and a brief summary of their purpose/role in your PR?

I added a crate list. I realize now that I forgot to mention that I added `libtock-rs`' examples to our build, which brought in an extra 2 or 3 crates.
